Lib/fontTools/ttLib/tables/_n_a_m_e.py

```diff
@@ -311,6 +311,16 @@ def addMultilingualName(self, names, ttFont=None, nameID=None,
 			if nameID is not None:
 				return nameID
 			nameID = self._findUnusedNameID()
+
+			nameID_win = self.findMultilingualName(names, windows=True, mac=False)
```

Comment:
I see what you're trying to do. I'm doubtful about the need: do we really need to handle this case? It seems that if you're building a font from scratch, you're best served by using the same settings for `mac` and `windows` for the entire build. In other words, if you're running into the situation where both mac and windows are requested but only one is present, it may be a symptom of an inconsistency elsewhere. I would be fine with `addMultilingualName` creating more name table entries than strictly necessary in this case.

That said, if we really do want to handle this in an optimal way, it should be structured slightly differently:

1. you're calling `findMultilingualName` for both mac and windows, regardless of whether they were requested.
2. you should check `nameID_win is not None and nameID_mac is None`, as `nameID_win` could legally be 0, which is also false. Same for the next if.
3. The second if should be an elif.
4. After the elif there should be an else in which the `nameID = self._findUnusedNameID()` call is done instead of before everything else, so it isn't called redunantly.

But I'd like some discussion whether we even want to handle this case.

Reply:
> the situation where both mac and windows are requested but only one is present

I don't think it's worth supporting that use case. Producing an extra nameID is not the end of the world, and as Just said it's probably a hint of some other inconsistency in the way the name table is generated.

Reply:
@justvanrossum @anthrotype Thank you very, very much for your feedback. I much appreciate the discussion and your valuable comments and thoughts. As discussed I went back one step, added a few more unittests and added some comments for explanation. Please see https://github.com/fonttools/fonttools/pull/2528/commits/e968e8fd7f39198e606e5605148599e5a50f2ffa